StripePayments/StripePayments/Source/PaymentHandler/STPPaymentHandler.swift

```diff
@@ -1475,13 +1475,25 @@ public class STPPaymentHandler: NSObject {
         return resultingUrl
     }
 
-    func _retryAfterDelay(retryCount: Int, delayTime: TimeInterval = 3, block: @escaping STPVoidBlock) {
-        DispatchQueue.main.asyncAfter(deadline: .now() + delayTime) {
+    /// Schedules delayed execution of a retry block and records a poll attempt.
+    /// - Parameters:
+    ///   - delay: Time interval to wait before execution (used as target interval for optimization)
+    ///   - pollingBudget: Budget tracker for polling attempts
+    ///   - block: Block to execute if budget allows
+    func pollAfterDelay(delay: TimeInterval = 1, pollingBudget: PollingBudget, block: @escaping () -> Void) {
+        let optimizedDelay = pollingBudget.recommendedDelay(targetInterval: delay)
+        DispatchQueue.main.asyncAfter(deadline: .now() + optimizedDelay) {
+            pollingBudget.recordPollAttempt()
```

Comment:
WDYT about moving this to the PollingBudget helper?  It'd make it impossible to accidentally not call `recordPollAttempt()`.

Reply:
We should also delete `delay`.

Reply:
Yeah that's a good idea, will do